staging/src/k8s.io/apiserver/pkg/authentication/authenticatorfactory/requestheader.go

```diff
@@ -28,9 +28,19 @@ type RequestHeaderConfig struct {
 	// ExtraHeaderPrefixes are the head prefixes to check (case-insentively) for filling in
 	// the user.Info.Extra.  All values of all matching headers will be added.
 	ExtraHeaderPrefixes []string
-	// VerifyOptionFn are the options for verifying incoming connections using mTLS.  Generally this points to CA bundle file which is used verify the identity of the front proxy.
-	// It may produce different options at will.
-	VerifyOptionFn x509request.VerifyOptionFunc
+	// CAContentProvider the options for verifying incoming connections using mTLS.  Generally this points to CA bundle file which is used verify the identity of the front proxy.
+	//	It may produce different options at will.
+	CAContentProvider CAContentProvider
 	// AllowedClientNames is a list of common names that may be presented by the authenticating front proxy.  Empty means: accept any.
 	AllowedClientNames []string
 }
+
+// CAContentProvider provides ca bundle byte content
+type CAContentProvider interface {
```

Comment:
How come this needs to be duplicated? Import cycle?

Reply:
> How come this needs to be duplicated? Import cycle?

Yeah, the authenticator shouldn't import an impl.